tests/Feature/Service/LanguageServiceTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Service;
+
+use Tests\TestCase;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use App\Service\Student\LanguageService;
+use Tests\Fixtures\Students;
+use Tests\Fixtures\Resumes;
+use App\Exceptions\StudentNotFoundException;
+use App\Exceptions\LanguageNotFoundException;
+use App\Exceptions\ResumeNotFoundException;
+
+use Illuminate\Support\Str;
+
+class LanguageServiceTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    protected $languageService;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->languageService = new languageService();
+    }
+
+    public function testLanguageServiceReturnsValidLanguageDetailsForStudentWithLanguages()
+    {
+        $student = Students::aStudent();
+
+        $studentId = $student->id;
+
+        $resume = Resumes::createResumeWithModality($studentId, 'frontend', ['tag1', 'tag2'], 'Presencial');
+        
+        $languages = [];
```

Comment:
Great test!
The only thing I would do is that from line 38 to 51, you transfer the method to a class for example: 

'Tests\Fixtures\Languages;'

and in this same line call it as: 
'$languages = Languages::createLanguages($resume, 2);'

Reply:
I'll try to work on it.
Working with fixtures is really more interesting.
Thanks for your suggestion!!

Reply:
Following your suggestion, I just created the LanguagesForResume fixture and applied it to the tests of both the controller and the service.